moveit_planners/ompl/ompl_interface/src/detail/ompl_constraints.cpp

```diff
@@ -385,4 +385,101 @@ std::shared_ptr<BaseConstraint> createOMPLConstraint(const moveit::core::RobotMo
     return nullptr;
   }
 }
-}  // namespace ompl_interface
\ No newline at end of file
+
+/******************************************
+ * Toolpath Constraint
+ * ****************************************/
+ToolPathConstraint::ToolPathConstraint(const moveit::core::RobotModelConstPtr& robot_model, const std::string& group,
+                                       unsigned int num_dofs, unsigned int steps)
+  : BaseConstraint(robot_model, group, num_dofs), num_steps_(steps)
+{
+  NearestNeighbors<geometry_msgs::msg::Pose>::DistanceFunction func =
+      std::bind(&poseDistance, std::placeholders::_1, std::placeholders::_2);
+
+  pose_nn_.setDistanceFunction(func);
```

Comment:
See the Effective Modern C++ book or [this](https://github.com/peter-can-write/cpp-notes/blob/master/prefer-lambdas-over-std::bind.md).
```suggestion
  pose_nn_.setDistanceFunction(
   [this](const geometry_msgs::msg::Pose& pose1, const geometry_msgs::msg::Pose& pose2) {
      return poseDistance(pose1, pose2);
   }
 )
```